tests/Backend/CommonPart2/JobPollDelayTest.php

```diff
@@ -0,0 +1,124 @@
+<?php
+
+namespace Keboola\Test\Backend\CommonPart2;
+
+use Keboola\Csv\CsvFile;
+use Keboola\Test\StorageApiTestCase;
+
+class JobPollDelayTest extends StorageApiTestCase
+{
+
+    public function setUp()
+    {
+        parent::setUp();
+        $this->_initEmptyTestBuckets();
+    }
+
+    public function testRetryJobPollDelays()
+    {
+        $client = new \Keboola\StorageApi\Client(array(
+            'token' => STORAGE_API_TOKEN,
+            'url' => STORAGE_API_URL,
+            'maxJobPollWaitPeriodSeconds' => 20,
+        ));
+
+        $csvFile = new CsvFile(__DIR__ . '/../../_data/languages.csv');
+        $tableId = $client->createTable($this->getTestBucketId(), 'languages', $csvFile);
+
+        $fileId = $client->uploadFile(
+            $csvFile->getPathname(),
+            (new \Keboola\StorageApi\Options\FileUploadOptions())
+                ->setNotify(false)
+                ->setIsPublic(false)
+                ->setCompress(false)
+                ->setTags(array('table-import'))
+        );
+        $start = microtime(true);
+        $job = $client->apiPost("storage/tables/{$tableId}/import-async", [
+            'dataFileId' => $fileId
+        ], false);
+        $job = $client->waitForJob($job['id']);
+        $duration = microtime(true) - $start;
+        // this standard job always takes 3 tries, so minimum sleep is 2 + 4 + 8 = 14 sec
```

Comment:
I am not sure if these tests will be stable

Reply:
Maybe you can just test that the function is called when injected

Reply:
yes, I tried to give the tests some leeway, but I agree.  I'll rework it as you say. 👍